### ISSUE TITLE
Make google-cloud-java forward compatible with Guava 23+

### ISSUE THREAD

**Issue by garrettjonesgoogle** [googleapis/google-cloud-java#3049]
I haven't personally tried this yet, but I have evidence that google-cloud-java is not currently forward compatible with Guava 23+. We need to make sure all code can run against both Guava 20 and Guava 23 (basically only depend on classes/methods that exist in the intersection of those versions).

**Comment by marshallpierce** [comment]
It does appear to be incompatible. I'm using Guava 26 and seeing:

```
ava.lang.NoSuchMethodError: com.google.common.util.concurrent.Futures.transform(Lcom/google/common/util/concurrent/ListenableFuture;Lcom/google/common/base/Function;)Lcom/google/common/util/concurrent/ListenableFuture;
	at com.google.api.core.ApiFutures.transform(ApiFutures.java:99)
	at com.google.cloud.firestore.UpdateBuilder.commit(UpdateBuilder.java:618)
	at com.google.cloud.firestore.WriteBatch.commit(WriteBatch.java:41)
	at com.google.cloud.firestore.DocumentReference.set(DocumentReference.java:171)
```

**Comment by jadekler** [comment]
Yikes. I think this only gets solved when we go to Java 8 (it seems guava 21.0 is last supported Java 7 version?).

**Comment by marshallpierce** [comment]
`ApiFutures` is in https://github.com/googleapis/api-common-java, which has already released a version (1.7.0) that doesn't use the deprecated Guava features. That is, in turn, depended on by `com.google.api:gax-grpc`, which has also released an updated version (1.30.0) that depends on api-common-java 1.7.0. So, using api-common 1.7.0 (possibly via gax-grpc 1.30.0) should do the trick.

**Comment by chingor13** [comment]
Guava upgraded to 26.0-android in https://github.com/googleapis/google-cloud-java/releases/tag/v0.71.0
